src/main/java/com/google/api/codegen/config/FlatteningAndRequiredFieldsConfigGenerator.java

```diff
@@ -25,20 +25,21 @@
 import java.util.Map;
 
 /**
- * Config generator for method parameter flattening.
+ * Config generator for method parameter flattening and required fields.
  */
-public class FlatteningConfigGenerator implements MethodConfigGenerator {
+public class FlatteningAndRequiredFieldsConfigGenerator implements MethodConfigGenerator {
```

Comment:
"FieldConfigGenerator" would be a lot shorter, but I think is sufficiently clear.